newIDE/app/src/CommandPalette/CommandManager.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+type CmdHandler = () => void | Promise<void>;
+
+export type GlobalCommand = {|
```

Comment:
See my other comment below, I think we could name this simply "Command". A command has a display text for the user, can be enabled or not, and has a function to be run. Seems good to me, but it's not global in any sense (it's the usage that you're making of it that determines if it's global or not).